### PR TITLE
Linux example

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

## Description
Brings linux example application based on this library

### CODE REVIEW

File: example/linux/main.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  if( cmd_has_option(argc, argv, "o") ) {
+    cmd_printf("This is o option\r\n");
+  } else {
+    return CMDLINE_RETCODE_INVALID_PARAMETERS;
+  }
+  return CMDLINE_RETCODE_SUCCESS;
+}
+volatile bool running = true;
+static int cmd_exit(int argc, char *argv[]) {
+  cmd_printf("Exiting cli.");
```
Comment (YifeiZuo01) [inline example/linux/main.c]: seems this print didn't get printed when cmd is exit.

Comment (jupe) [reply example/linux/main.c]: fid you try with ncurses or without? most probably ncurses exit causes that..

Comment (YifeiZuo01) [reply example/linux/main.c]: without `#define EXAMPLE_USE_NCURSES 0`
when `#define EXAMPLE_USE_NCURSES 1`, `./cli`, this `tr_info("write 'help' and press ENTER");` get printed after `exit`

Comment (jupe) [reply example/linux/main.c]: removed print since it's unnecessary

**Pull request merged**